### DISCUSSION TITLE
🚀 Fleetbase v0.1.5 Release Alert! 🚀

### DISCUSSION THREAD

**Discussion by roncodes** [fleetbase/fleetbase#77] [Announcements]
Hello Fleetbase Family,

Exciting news ahead! We’re supercharged to introduce a major update - presenting Fleetbase v0.1.5! We're always striving to augment your Fleetbase experience, and our latest release is no exception. With a smorgasbord of new features, improvements, and bug fixes, this version promises to make your journey with us even more rewarding.

Firstly, we’re excited to unveil an alpha version of a significant addition to our internal API for building integrated Fleetbase extensions: the `UniverseService`. Designed to improve the seamless interactions between extensions and the console, this service provides an interface for a whole new range of features. From adding menu items and pages throughout the Console to facilitating custom settings in user, admin, and organization sections - the `UniverseService` has got you covered. Moreover, it opens up the door for extensions to trigger and listen to events throughout the console. Imagine customizing an OrderModel upon creation - now a reality with our `UniverseService`. While this is just the Alpha version with a limited feature set, we're already working on expanding its capabilities. Future releases of the `UniverseService` will enable powerful customizations and seamless features via "Extensions". This is only the beginning, and the best is yet to come.

Parallel to the `UniverseService`, we've updated another internal API - the `setupExtension` hook on the `engine.js`. This hook simplifies the registration of your extension and more. It presents the console application instance, your extension engine instance, and the universe service itself to you in one place.

_Here's a little taste of how you can utilize it:_

```js
import Engine from '@ember/engine';
import loadInitializers from 'ember-load-initializers';
import Resolver from 'ember-resolver';
import config from './config/environment';
import services from '@fleetbase/ember-core/exports/services';
import CustomSettingsComponent from './components/custom-settings';
import CustomAdminSettingsComponent from './components/custom-admin-settings';

const { modulePrefix } = config;
const externalRoutes = ['console', 'extensions'];

export default class MyExtensionEngine extends Engine {
    modulePrefix = modulePrefix;
    Resolver = Resolver;
    dependencies = {
        services,
        externalRoutes,
    };
    setupExtension = function (app, engine, universe) {
        // - Register Extension in the Console Header -
        // the first and secon parameter is the link title, and the route
        // the route is dynamically generated based on the extension name, you can see
        // what Fleetbase has determined as the extension route by seeing `app/router.js` in console dir
        // If you want to specify your own route you can do so in your extension `package.json` under `fleetbase.route`
        // like so "fleetbase": { "route": "my-extension" } 
        // important to note all routes will be prefixed by `console.` as this is where extensions live
        // `priority` determines the position in the console header
        // `icon` will render the selected icon
        universe.registerHeaderMenuItem('My Extension', 'console.my-extension', { icon: 'flask', priority: 0 });

        // - Register Custom Settings in Organization Dropdown Menu -
        // The route will always point to a virtual point with a component to render
        // There will be more details about this in the updated docs
        universe.registerOrganizationMenuItem('My Extension Settings', {
            icon: 'flask',
            index: 3,
            slug: 'my-extension',
            component: CustomSettingsComponent,
            onClick: (menuItem) => {
                return universe.transitionMenuItem('console.settings.virtual', menuItem);
            },
        });

        // - Register a Custom Menu Item in the Settings Side Navigation -
        universe.registerSettingsMenuItem('My Extension', {
            slug: 'my-extension',
            icon: 'flask',
            component: CustomSettingsComponent,
        });

        // - Create a Custom Menu Panel in the Admin Section -
        universe.registerAdminMenuPanel(
            'My Extension Configuration',
            [
                {
                    title: 'Settings',
                    icon: 'cog',
                    component: CustomAdminSettingsComponent
                },
                {
                    title: 'Help',
                    icon: 'help',
                    component: CustomAdminSettingsComponent,
                },
                {
                    title: 'Subscriptions',
                    icon: 'money-bill-wave',
                    component: CustomAdminSettingsComponent,
                },
            ],
            {
                slug: 'my-extension',
            }
        );
    };
}

loadInitializers(MyExtensionEngine, modulePrefix);
```

Beyond these robust internal API updates, we've ironed out some bugs to make your experience smoother. Of particular note, we've remedied a glitch in Fleet Ops that disrupted the display route on post-order creation. If you've encountered this bug, the update should rectify the issue.

In conclusion, we're thrilled about the enhancements brought about by v0.1.5 and can't wait for you to experience them. As always, we are eager to hear your feedback. Enjoy the improved Fleetbase experience, and stay tuned for more updates!
